lib/assignable_values/active_record/restriction/belongs_to_association.rb

```diff
@@ -42,7 +42,7 @@ def previously_saved_value(record)
 
         def current_value(record)
           value = record.send(property)
-          value = record.send(property, true) if (value && value.id) != association_id(record)
+          value = value.reload if !value.nil? && !value.new_record? && (value && value.id) != association_id(record)
```

Comment:
Thanks for working on a fix, we definitely want to make this work on 5.1.

However, I don't think this particular fix is correct. I believe the suggested replacement for `record.send(property, true)` would be to call `record.reload`, not `value.reload`. But we absolutely cannot do that because of all the side effects (i.e. loss of validation errors etc).

I think it might be possible to do something along the lines of `record.association(property).reload`.

To be quite honest, I have no idea what this line is actually protecting us from in the first place. How would one get into a situation where `record.associated.id != record.associated_id`? Maybe we can simply drop it? @triskweline, do you remember?

Reply:
I don't know about any situation where `record.associated.id != record.associated_id` with ActiveRecord.

I propose to remove it and see if a test fails.

Reply:
At least one spec does a `should_receive(..., true)`. But I agree, let's remove that one and see if there are other fails.

Reply:
the tests seem to be fine, i also tried this fix in the project i am currently working on, also worked without the reloading line.